src/lib.rs

```diff
@@ -144,7 +144,7 @@ fn is_ty_no_move(tcx: &ty::ctxt, t: ty::Ty) -> bool {
                 }
                 true
             },
-            ty::TyRef(..) | ty::TyRawPtr(..) => false, // don't recurse down ptrs
+            ty::TyBox(..) | ty::TyRef(..) | ty::TyRawPtr(..) => false, // don't recurse down ptrs
```

Comment:
Box is an owning ref, the other two are not. The reason we don't recurse here is that the other two are not owned references so cannot change the underlying ownership of the thing being pointed to.

On the other hand, box is owned, and moving it is a change of ownership.


This crate considers a move to be any change in ownership, not just an in-memory move.